### PR TITLE
Update to Gradle 4.0.2

### LINKED ISSUE #858: Update to Gradle 4.0

### PR DESCRIPTION
**Pull request by AustinShalit**
[//]: # (Please ensure that the "Allow edits from maintainers" checkbox is checked. Thanks!)

closes #858

### CODE REVIEW

File: core/src/test/java/edu/wpi/grip/core/CoreSanityTest.java [hunk 1]
```diff
@@ -27,7 +27,8 @@ public CoreSanityTest() {
         ManualPipelineRunner.class,
         SubtractionOperation.class,
         Main.class,
-        CoreCommandLineHelper.class
+        CoreCommandLineHelper.class,
+        Pipeline.class
```
Comment (AustinShalit) [inline core/src/test/java/edu/wpi/grip/core/CoreSanityTest.java]: @JLLeitschuh Can you take a look and see why I had to ignore Pipeline to get it to build?

Comment (JLLeitschuh) [inline core/src/test/java/edu/wpi/grip/core/CoreSanityTest.java]: The test wouldn't pass or the compiler would fail??

Comment (AustinShalit) [inline core/src/test/java/edu/wpi/grip/core/CoreSanityTest.java]: Tests would not pass.

File: gradle.properties [hunk 2]
```diff
@@ -1,3 +1,3 @@
-org.gradle.parallel=true
-org.gradle.configureondemand=true
-org.gradle.cache.tasks=true
+org.gradle.parallel=false
+org.gradle.configureondemand=false
+org.gradle.cache.tasks=false
```
Comment (JLLeitschuh) [inline gradle.properties]: Sure, we can do this, but why is this necessary? The gradle 4.0 default is now to run with parallel true.

Comment (AustinShalit) [inline gradle.properties]: I just updated and removed the file entirely.

Comment (JLLeitschuh) [inline gradle.properties]: I saw, thats fine. `org.gradle.configureondemand=true` can speed up the build a bit but its not really that important in this build because its so small.

File: core/src/test/java/edu/wpi/grip/core/CoreSanityTest.java (hunk 1 above)
Comment (JLLeitschuh) [inline core/src/test/java/edu/wpi/grip/core/CoreSanityTest.java]: Can you paste the stack trace?

Comment (AustinShalit) [inline core/src/test/java/edu/wpi/grip/core/CoreSanityTest.java]:
```
edu.wpi.grip.core.CoreSanityTest > testNulls FAILED
    junit.framework.AssertionFailedError: Error in automated nulls test of class edu.wpi.grip.core.Pipeline
    If the class is better tested explicitly, you can add testNulls() to edu.wpi.grip.core.PipelineTest

        Caused by:
        junit.framework.AssertionFailedError: Can't find or create a sample instance for type 'edu.wpi.grip.core.Step'; please provide one using NullPointerTester.setDefault()
```

Comment (JLLeitschuh) [inline core/src/test/java/edu/wpi/grip/core/CoreSanityTest.java]: That makes sense. Uhhh.
In the code below this where all the `setDefault`'s are add:
```java
setDefault(Step.class, new MockStep());
```
That should solve this problem.

Comment (AustinShalit) [inline core/src/test/java/edu/wpi/grip/core/CoreSanityTest.java]: Thanks!

**Comment by AustinShalit** [comment]
@JLLeitschuh Ready for review
